airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -42,6 +43,31 @@ class SlackStream(HttpStream, ABC):
     url_base = "https://slack.com/api/"
     primary_key = "id"
     page_size = 100
+    # We need different retry behaviour for different response types, implement
+    # own retry counter.
+    retry_attempts = 0
+    max_retry_attempts = 5
+
+    @property
+    def max_retries(self) -> Union[int, None]:
+        self.retry_attempts = 0
+        # Return None for unlimited retry attempts. This stream implements its
+        # own retry counter based on response type.
+        return None
+
+    def should_retry(self, response: requests.Response) -> bool:
+        if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+            """
+            According to this https://api.slack.com/docs/rate-limits#headers
+            Slack has no any limitation / restriction on API usage (except of
+            posting message over API when app can be blocked), we should only
+            follow the rules and evaluate delay given by "Retry-After" header.
+            Do not give up retry attempts in case of 429 error.
+            """
+            return True
```

Comment:
As I understand that when we receive 429 errors, we will always make a sleep for code and one more try for current request. Perhaps it is worth adding, if not the number of retries for one request, then the maximum time, so that an infinite time loop does not turn out? WDYT?

Reply:
Make sense, updated PR